junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/descriptor/TestFactoryTestDescriptor.java

```diff
@@ -88,8 +88,12 @@ protected void invokeTestMethod(JupiterEngineExecutionContext context) {
 
 			try {
```

Comment:
Should be using try-with-resource syntax, like:
```
AtomicInteger index = new AtomicInteger();
try (Stream<DynamicTest> stream = toDynamicTestStream(testExtensionContext, testFactoryMethodResult)) {
    stream.forEach(dynamicTest -> registerAndExecute(dynamicTest, index.incrementAndGet(), listener));
}
```
No need to close the stream manually, if exception are thrown.